build/index.js

```diff
@@ -0,0 +1,127 @@
+/* global Promise */
+/* jshint node: true */
+
+var xcodeUtil = require('../../devkit-core/modules/native-ios/lib/xcodeUtil'),
+  updatePlist = require('../../devkit-core/modules/native-ios/lib/updatePlist'),
+  Rsync = require('rsync'),
+  module_config = require("../ios/config"),
+  path = require("path");
+
+var isHeaderFile = function(filename) {
+  return (/\.h(pp)?$/).test(filename);
+};
+
+var isSourceFile = function(filename) {
+  return (/\.(c(pp)?)|mm?$/).test(filename);
+};
+
+exports.onBeforeBuild = function (api, app, config, cb) {
+  'use strict';
+
+  var widgetID = app.manifest.ios.widgetID || 'widget',
+    widgetGroup = app.manifest.ios.widgetGroup || 'group.' + config.bundleID,
+    manifest = app.manifest,
+    version = manifest.ios && manifest.ios.version || manifest.version || '0.0.0',
+    buildNumber = manifest.ios && manifest.ios.buildNumber || version;
+
+  cb();
+
+  if (config.target !== 'native-ios') {
+    return;
+  }
+
+  api.streams.registerFunction('ios-build-widget', function () {
+    var xcodeProjectPath = config.xcodeProjectPath,
+      groupKey, target;
+
+    return Promise
+      .resolve()
+      .then(function () {
+        var rsync = new Rsync()
+          .flags('a')
+          .set('delete-before')
+          .source(path.resolve(__dirname, '../ios/widget'))
+          .destination(xcodeProjectPath);
+
+        return Promise.fromNode(rsync.execute.bind(rsync));
+      })
+      .then(function () {
+        return xcodeUtil.getXcodeProject(xcodeProjectPath);
+      })
+      .then(function (xcodeProject) {
+        var project = xcodeProject._project,
+          firstProject = project.getFirstProject().firstProject,
+          firstTarget = project.getFirstTarget().firstTargetUuid;
```

Comment:
Okay, Can we always assume the position ? Is there a better way to figure which is the widget block ?